src/storage/adapter/mod.rs

```diff
@@ -22,3 +27,32 @@ pub trait StorageAdapter: std::fmt::Debug {
     /// Removes a record from the storage.
     async fn remove(&mut self, key: &str) -> crate::Result<()>;
 }
+
+#[derive(Debug, Default)]
+/// A storage adapter that stores data in memory.
+pub struct Memory(HashMap<String, String>);
+
+#[async_trait::async_trait]
+impl StorageAdapter for Memory {
```

Comment:
Can you put memory in a dedicated module please ?

Reply:
I left it in mod as its sort of a testing/fallback adapter, but yea i will move it